migration.rb

```diff
@@ -0,0 +1,60 @@
+class MigrateWf
+  attr_accessor :failed_account_ids, :req_counter, :start_time, :options
+
+  def initialize
+    @failed_account_ids = []
+    @req_counter = 0
+    @start_time = Time.now.to_i
+    @options = { payload_type: 'workflow_sync', meta: { 'source' => 'search' } }
+  end
+
+
+  def check_redis
+    BaseRedis.key_exists?("STOP_WF_SEARCH_MIGRAION")
+  end
+
+  def ensure_rpm
+    time_elapsed = Time.now.to_i - @start_time
+    if time_elapsed < 60
+      puts "sleeping for #{60 - time_elapsed} - account #{Account.current.id}"
+      sleep(60 - time_elapsed)
+    end
+    @req_counter = 0
+    @start_time = Time.now.to_i
+  end
+
+  def central_push(wf)
+    payload = wf.collect_properties_v2(@options)
+    payload[:event_uuid] = wf.event_uuid || wf.generate_event_id
+    payload = { params: payload, version: 2 }
+    res = FreshPipeReSync.new.perform(payload)
+    @req_counter += 1 if res
+  end
+
+  def migrate_wf(shard_names, rpm)
+    shard_names.each do |shard_name|
+      Sharding.run_on_shard(shard_name) do
+        AccountIterator.each(batch_size: 200) do |account|
```

Comment:
Incase of restart, we should resume from where it last completed based on redis value as we discussed?